### PR TITLE
PLAT-140834: Add UI Test for VirtualGridList

### PR DESCRIPTION
**Pull request by dongsuwon**
Enact-DCO-1.0-Signed-off-by: Dongsu Won (dongsu.won@lgepartner.com)

### Checklist
* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- Need to add VirtualGridList ui-test.
- Need to add button to add item.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Add VirtualGridList ui-test.
- Add button to add item.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]:# (Related issues, references)
PLAT-140834

### CODE REVIEW

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28987-specs.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		// Step 3-2: 5-way Left to Item 16.
+		for (let i = 0; i < 8; i++) {
+			Page.spotlightLeft();
+			Page.delay(200);
+		}
+		Page.delay(500);
+		// Step 3-2 Verify: Spotlight is on item 16.
+		expectFocusedItem(16);
+		const curScrollThumbPosition = Page.scrollThumbPosition();
+
```
Comment (jeonghee27) [inline tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28987-specs.js:27]: How about verifying "curScrollThumbPosition > 0" or "curScrollThumbPosition != initialThumbPoisition"

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28990-specs.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		// Step 2 Verify: Spotlight is on the 2nd item.
+		expectFocusedItem(1);
+
+		// Step 3: Keep pressing 5-way Down until the list reaches to the bottom.
+		for (let i =  0; i < 20; i++) {
+			Page.spotlightDown();
+			Page.delay(200);
+		}
+		// Step 3-1 Verify: Spotlight is on the 101th item with 'Image 100'.
+		expectFocusedItem(100);
```
Comment (jeonghee27) [inline tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28990-specs.js:24]: I think "Step 3-2 Verify: The 101th item with 'Image 100' *sticks to the bottom in the list*" is not verified by below code.

How about add verifying the scrollPoisition. 
```suggestion
		expectFocusedItem(100);
		expect(Page.scrollThumbPosition()).to.equal('1');
```
Or, you can using itemOffset.

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28982-specs.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const {expectFocusedItem} = require('../../VirtualList-utils');
+
+describe('qa-VirtualGridList translate mode', function () {
+	beforeEach(function () {
+		Page.open();
+	});
+
+	it('should spotlight is on the bottom/top in same column via page up/down key [GT-28982]', function () {
+		// Step 2: Uncheck the "Native Scrolling" CheckboxItem.
+		Page.buttonModeChange.moveTo();
```
Comment (jeonghee27) [inline tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28982-specs.js:11]: I wonder, there is same TC for native scroll. 
If not, how about copy code and attach below on this file for native?

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28987-specs.js (hunk 1 above)
Comment (dongsuwon) [reply tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28987-specs.js:27]: It compared scrollthumb's positon on item16 and item14. This value is not 0.
When press 5-way right on item 16, the scrollthumb's position should not change.

File: tests/ui/apps/VirtualList/VirtualGridList/VirtualGridList-View.js [hunk 4]
```diff
@@ -111,7 +111,13 @@ class app extends Component {
...
-		this.setState({numItems: value});
+		this.setState({numItems: Number(value)});
+		updateData(value);
+	};
+
+	onAddNumItem = () => {
+		const value = this.state.numItems + 1;
+		this.setState({numItems: Number(value)});
 		updateData(value);
 	};
```
[the remark this answers is not in the document]
Comment (dongsuwon) [reply tests/ui/apps/VirtualList/VirtualGridList/VirtualGridList-View.js]: Sugestted chagne.

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28990-specs.js (hunk 2 above)
Comment (dongsuwon) [reply tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28990-specs.js:24]: Sugestted chagne.

File: tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28982-specs.js (hunk 3 above)
Comment (dongsuwon) [reply tests/ui/specs/VirtualList/VirtualGridList/qa-VirtualGridList/VirtualGridList-GT-28982-specs.js]: Sugestted chagne.

**Comment by dongsuwon** [comment]
Jenkins Passed:
/job/enact-ui-tests/1633/

**Review by jeonghee27 — APPROVED** [review]
LGTM

**Pull request merged**